src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -259,6 +259,26 @@ Copyright (c) .NET Foundation. All rights reserved.
       <Output TaskParameter="AssembliesToPublish" ItemName="ResolvedAssembliesToPublish" />
     </ResolvePublishAssemblies>
   </Target>
+
+  <Target Name="FilterPublishSatelliteResources" AfterTargets="RunResolvePublishAssemblies"
+          Condition="'$(SatelliteResourceLanguages)' != ''">
+
+    <ItemGroup>
+      <PublishSatelliteResources Include="@(ResolvedAssembliesToPublish)"
+                                 Condition="'%(ResolvedAssembliesToPublish.AssetType)' == 'resources'" />
+    </ItemGroup>
+    
+    <JoinItems Left="@(PublishSatelliteResources)" LeftKey="Culture" LeftMetadata="*"
```

Comment:
The one remaining JoinItems still shows up as relatively slow. We probably want to see about speeding it up. Given that this only kicks in with SatelliteResourceLanguages set, it is probably OK.